presentation-api/controlling-ua/PresentationConnection_onclosed-manual.html

```diff
@@ -21,28 +21,36 @@
 <script>
   setup({explicit_timeout: true});
   var startPresentation = function () {
+
     document.getElementById('presentBtn').disabled = true;
-    async_test(function(t) {
+
+    promise_test(function (t) {
       var request = new PresentationRequest(presentationUrls);
-      request.start()
-        .then(function(connection) {
+
+      return request.start()
+        .then(function (connection) {
 
           // Enable timeout again, cause no user action is needed from here.
-          t.step_timeout(function() {
+          t.step_timeout(function () {
               t.force_timeout();
               t.done();
           }, 5000);
 
           assert_true(connection instanceof PresentationConnection, 'the connection is setup');
-          connection.onclose = t.step_func_done(function(evt) {
+
+          connection.onclose = function (evt) {
```

Comment:
If you want the test to wait for onclose to be called, you'll need to return some promise that doesn't resolve until onclose is called. EventWatcher might help with that, something like (untested):
```js
const w = new EventWatcher(t, connection, 'close');
const p = w.wait_for('close');
connection.close();
return p;
}.then(function (evt) {
  // assertions
});